spec/middleware-1.0.0-draft.1.md

```diff
@@ -0,0 +1,309 @@
+# OWIN Middlewares
+
+**Version**
+1.0.0-draft.1
+
+**Authors**
+Sebastien Lambla (Caffeine IT)
+
+**License**
+[Creative Commons Attribution 3.0 Unported License][CCLicense]
+
+**Last updated**
+1 October 2015
+
+## Abstract
+
+The [OWIN Specification][owin-spec] defines a standard interface between .NET web servers
+and web applications. This document specifies additional standard interfaces
+for reusable OWIN Middleware components.
+
+## Status
+
+This is a working draft.
+
+This document is a submission to the OWIN Working Group and is valid for a
+period of six months and may be updated, replaced, or obsoleted by other
+documents at any time. It is inappropriate to use working drafts as reference
+material or to cite them other than as "work in progress"
+
+This working draft will expire on 1 March 2016.
+
+## Copyright
+
+Copyright (c) 2014 the persons identified as the document authors. The work
+is licensed as per the
+[Creative Commons Attribution 3.0 Unported License][CCLicense] in effect on the
+date of publication of this document.
+
+## Table of Content
+
+1. [Introduction](#1-introduction)  
+  1.1. [Requirements Notation](#11-requirements-notation)
+2. [Definitions](#2-definitions)
+3. [Middleware](#3-middleware)  
+  3.1. [Examples](#31-examples)  
+  3.2. [Middleware Operation](#32-middleware-operation)  
+  3.3. [Signature](#33-signature)  
+  3.4. [Code Sample](#34-code-sample)
+4. [Middleware Registration](#4-middleware-registration)  
+  4.1. [Signature](#41-signature)  
+    4.1.1. [Registration Sample](#411-registration-sample)  
+  4.2. [Discovery and Registration of Middleware](#42-discovery-and-registration-of-middleware)  
+    4.2.1. [Extension Method Sample](#421-extension-method-sample)  
+5. [Versioning](#5-versioning)
+6. [Acknowledgements](#6-acknowledgements)
+
+## 1. Introduction
+
+The [OWIN] specification was created to decouple web servers and web
+applications on .NET platforms by creating a standardised abstract signature
```

Comment:
Should "web servers" and "web applications" here be bolded, title-cased, and/or linked to the defnitions below?